_posts/2024-01-12-kasten-openshift-monitoring.markdown

```diff
@@ -0,0 +1,815 @@
+```
+
+you should have this complete response
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": [
+      {
+        "metric": {
+          "__name__": "version",
+          "endpoint": "web",
+          "instance": "10.130.2.35:8080",
+          "job": "prometheus-example-app",
+          "namespace": "ns1",
+          "pod": "prometheus-example-app-6884c645b9-gl2mm",
+          "prometheus": "openshift-user-workload-monitoring/user-workload",
+          "service": "prometheus-example-app",
+          "version": "v0.4.1"
+        },
+        "value": [
+          1704813472.324,
+          "1"
+        ]
+      }
+    ]
+  }
+}
+``` 
+
+# Apply to kasten 
+
+## Allow openshift monitoring to scrape kasten-io namespace
+
+First we need to allow request coming from the openshift-user-workload-monitoring namespace, for that we need to add a network policy in the kasten-io namespace.
+
+```
+cat<<EOF |oc create -f -
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  labels:
+    app: k10    
+  name: allow-openshift-monitoring
+  namespace: kasten-io  
+spec:
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          name: openshift-user-workload-monitoring    
+  - ports:
+    - port: http
+      protocol: TCP
+  podSelector:
+    matchLabels:
+      release: k10      
+  policyTypes:
+  - Ingress
+EOF
+```
+
+Check you can request the metrics endpoint in kasten from the `openshift-user-workload-monitoring` namespace by shelling in one of his prometheus pod.
+
+```
+oc rsh -n openshift-user-workload-monitoring prometheus-user-workload-0
+curl http://catalog-svc.kasten-io.svc.cluster.local:8000/metrics
+```
+
+If you get an answer it means that you allow communication to kasten-io from openshift-user-workload-monitoring.
+
+## Create a Service Monitor 
+
+Now let's create a service monitor for the catalog 
+
+```
+cat<<EOF | oc create -f -
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    k8s-app: kasten-io
+  name: catalog
+  namespace: kasten-io
+spec:
+  endpoints:
+  - interval: 30s
+    port: http
+    scheme: http
+  selector:
+    matchLabels:
+      component: catalog
+EOF
+```
+
+After some time you should be able to query the catalog_actions_count metrics from the openshift monitoring service 
+
+```
+SECRET=`oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token | head -n 1 | awk '{print $1 }'`
+TOKEN=`echo $(oc get secret $SECRET -n openshift-user-workload-monitoring -o json | jq -r '.data.token') | base64 -d`
+THANOS_QUERIER_HOST=`oc get route thanos-querier -n openshift-monitoring -o json | jq -r '.spec.host'`
+curl -X GET -kG "https://$THANOS_QUERIER_HOST/api/v1/query?" --data-urlencode "query=catalog_actions_count" -H "Authorization: Bearer $TOKEN" |jq
+```
+
+You should get an answer like this one (Unless you did not create any kasten action): 
```

Comment:
```suggestion
You should get an answer like this one (Unless you did not trigger any Kasten actions): 

{: .alert-info }
If you don't see a response similar to the one below, it's likely because you have not triggered any actions (e.g. Backup, export, restore, etc) within Kasten. If you do so, then re-run the above commands, you should hopefully see a similar response.
```